lib/libc/minimal/include/sys/stat.h

```diff
@@ -1,9 +1,51 @@
 /*
- * Copyright (c) 2018 Linaro Limited
+ * Copyright (c) 2018 Intel Corporation
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#ifndef ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_SYS_STAT_H_
+#define ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_SYS_STAT_H_
 
-/* Dummy sys/stat.h to fulfill the requirements of certain libraries.
- */
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <kernel.h>
```

Comment:
Why keep this? It's not required for the definitions later in the header, and seems a bit of a layering violation for the C library to be pulling in Zephyr kernel headers.

Reply:
> Why keep this?

In general, to not introduce potential regressions. If the change says "move file", it should move file with minimal changes (I had to update header guards, as you imagine), instead of trying to squeeze other "helpful" stuff, which someone later may debug, pulling their hair out.

> and seems a bit of a layering violation for the C library to be pulling in Zephyr kernel headers.

Well, for an I/O related header like sys/stat.h there's nothing unusual to include headers related to OS I/O implementation (like kernel headers). Who knows, maybe something which includes this header relies later on those definitions to be available. Issue that or not, that's not what I fix here. Again, this started as a simple 1-line patch of moving 1 line from one file to another. This caused 4 levels of cascading failures, and turned out to required 5 patch to make CI pass. There's nothing more here to fix in *this PR*. (And a lot to fix in many more PRs.)

Reply:
> In general, to not introduce potential regressions. 

Sounds to me like any such code not caught by sanitycheck was buggy in the first place by using things it couldn't know for sure were included.

Why not remove it in a separate commit in the same PR and see what sanitycheck says?

Reply:
> not caught by sanitycheck

And out of tree users will have to do their own testing, of course. If I previously had code that pulled in fcntl.h and relied that to pull in zephyr's gpio.h, and someone changed things around so my build breaks, I think that's my fault, not theirs.

Reply:
> Sounds to me like any such code not caught by sanitycheck was buggy in the first place by using things it couldn't know for sure were included.

... And yet I still can't fix all the wrongs on Zephyr in one PR, instead trying to do that PR by PR. The sooner one PR merged, the sooner more fixes come in. Anyway, back to topic,  which color do we like them, bikesheds? Me likes green.

> Why not remove it in a separate commit in the same PR and see what sanitycheck says?

I can do that by personal request of yours, and with full understanding that it's outside the scope of this PR ;-).